cmplibrary/src/test/java/com/sourcepoint/cmplibrary/SpConsentLibImplTest.kt

```diff
@@ -121,15 +122,14 @@ class SpConsentLibImplTest {
             )
         }
 
-        verify { logger.i(any(), "${OTT.code}") }
+        verify { logger.i(any(), LEGACY_OTT.name) }
```

Comment:
Are you testing the `logger`?

Reply:
this test check that whenever the user select a MessageType from the public api `loadPrivacyManager`, the SDK is generating the correct url type for the pm

Reply:
the logger is a spy object

Reply:
Not sure if I understood you well. If we remove the log from `loadPrivacyManager` method, will the test fail?

Reply:
yes

Reply:
I think we need to move away from using the loggers as spy objects. It's not a robust solution in my opinion, because it relies on a side effect (the loggers being called). 

In this specific case we can:
* extract the logic smaller functions and test those via unit tests
* spy on the actual webview object to make sure the right URL was passed to load url via unit/integration tests 
* assert the right message renders via UI tests.

Since the SDK is already relying on loggers for testing, we should address that in a different PR.

Reply:
I see what you mean, this test here is an integration test that runs locally without the need of instancing an Emulator (which requires to build a test apk and then start an emulator and this requires at least 30-40 sec).
This test itself runs in 2 sec instead of a UI or Instrumentation test that requires 15 sec